lib/system/assertions.nim

```diff
@@ -0,0 +1,86 @@
+include "system/assertions_helpers"
```

Comment:
This makes me nervous. Is this included only once, or does this lead to code duplication?

Reply:
It did lead to code duplication, so I redid that stuff.

File `system/assertions_helpers` (which had the duplicated code) is now gone.
Now there's a single file `system/fatal.nim` which is included [here](https://github.com/nim-lang/Nim/pull/10597/files#diff-e02ab787ce7af9352584fe91e2bf2b6dR3) and [here](https://github.com/nim-lang/Nim/pull/10597/files#diff-5cfdfeab79c2d2b7bbc78af7060c38c3R2884).

IMO, the PR is new much cleaner than in the original version.